project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/UserDataFixture.php

```diff
@@ -31,48 +36,102 @@ class UserDataFixture extends AbstractReferenceFixture implements DependentFixtu
+     */
+    private $customerDataFactory;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Customer\UserDataFactoryInterface
+     */
+    private $userDataFactory;
+
     /**
      * @param \Shopsys\FrameworkBundle\Model\Customer\CustomerFacade $customerFacade
-     * @param \Shopsys\ShopBundle\DataFixtures\Demo\UserDataFixtureLoader $loaderService
      * @param \Faker\Generator $faker
      * @param \Doctrine\ORM\EntityManagerInterface $em
      * @param \Shopsys\FrameworkBundle\Component\String\HashGenerator $hashGenerator
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrameworkBundle\Model\Customer\CustomerDataFactoryInterface $customerDataFactory
+     * @param \Shopsys\FrameworkBundle\Model\Customer\UserDataFactoryInterface $userDataFactory
      */
     public function __construct(
         CustomerFacade $customerFacade,
-        UserDataFixtureLoader $loaderService,
         Generator $faker,
         EntityManagerInterface $em,
-        HashGenerator $hashGenerator
+        HashGenerator $hashGenerator,
+        Domain $domain,
+        CustomerDataFactoryInterface $customerDataFactory,
+        UserDataFactoryInterface $userDataFactory
     ) {
         $this->customerFacade = $customerFacade;
-        $this->loaderService = $loaderService;
         $this->faker = $faker;
         $this->em = $em;
         $this->hashGenerator = $hashGenerator;
+        $this->domain = $domain;
+        $this->customerDataFactory = $customerDataFactory;
+        $this->userDataFactory = $userDataFactory;
     }
 
     /**
      * @param \Doctrine\Common\Persistence\ObjectManager $manager
      */
     public function load(ObjectManager $manager)
     {
+        /** @var \Shopsys\FrameworkBundle\Model\Country\Country[] $countries */
         $countries = [
-            $this->getReference(CountryDataFixture::COUNTRY_CZECH_REPUBLIC),
-            $this->getReference(CountryDataFixture::COUNTRY_SLOVAKIA),
+            self::FIRST_DOMAIN_ID => $this->getReference(CountryDataFixture::COUNTRY_CZECH_REPUBLIC),
+            self::SECOND_DOMAIN_ID => $this->getReference(CountryDataFixture::COUNTRY_SLOVAKIA),
         ];
-        $this->loaderService->injectReferences($countries);
-
-        $customersData = $this->loaderService->getCustomersDataByDomainId(Domain::FIRST_DOMAIN_ID);
-
-        foreach ($customersData as $customerData) {
-            $customerData->userData->createdAt = $this->faker->dateTimeBetween('-1 week', 'now');
-
-            $customer = $this->customerFacade->create($customerData);
 
-            if ($customer->getId() === 1) {
-                $this->resetPassword($customer);
-                $this->addReference(self::USER_WITH_RESET_PASSWORD_HASH, $customer);
+        $fakerSeed = self::FAKER_SEED;
+
+        $domainConfigs = $this->domain->getAll();
+        foreach ($domainConfigs as $domainConfig) {
+            $this->faker->seed($fakerSeed++);
```

Comment:
is it somehow possible to use 
locale providers?
https://github.com/fzaninotto/Faker/tree/master/src/Faker/Provider/cs_CZ
https://github.com/fzaninotto/Faker/tree/master/src/Faker/Provider/sk_SK
maybe it would bring only more problems but if it would look great :)